guides/source/action_controller_overview.md

```diff
@@ -617,110 +705,85 @@ class ClientsController < ApplicationController
-    @comment = Comment.new(author: cookies[:commenter_name])
-  end
+* [`ActionDispatch::Session::CookieStore`][] - Stores everything on the client.
+* [`ActionDispatch::Session::CacheStore`][] - Stores the data in the Rails cache.
+* [`ActionDispatch::Session::ActiveRecordStore`][activerecord-session_store] -
+  Stores the data in a database using Active Record (requires the
+  [`activerecord-session_store`][activerecord-session_store] gem)
+* A custom store or a store provided by a third party gem
 
-  def create
-    @comment = Comment.new(comment_params)
-    if @comment.save
-      flash[:notice] = "Thanks for your comment!"
-      if params[:remember_name]
-        # Remember the commenter's name.
-        cookies[:commenter_name] = @comment.author
-      else
-        # Delete cookie for the commenter's name cookie, if any.
-        cookies.delete(:commenter_name)
-      end
-      redirect_to @comment.article
-    else
-      render action: "new"
-    end
-  end
-end
-```
+For most session stores, the unique Id in the cookie is used to look up session data on the server (e.g. a database table). Rails does not allow you to pass the session ID in the URL as this is less secure.
 
-Note that while for session values you can set the key to `nil`, to delete a cookie value you should use `cookies.delete(:key)`.
+The `CookieStore` is the default and recommended session store. It stores all session data in the cookie itself (the ID is still available to you if you need it). The cookie data is cryptographically signed to prevent tampering. It's also encrypted so anyone with access to it can't read its contents. The `CookieStore` is lightweight and does not require any configuration to use in a new application.
 
-Rails also provides a signed cookie jar and an encrypted cookie jar for storing
-sensitive data. The signed cookie jar appends a cryptographic signature on the
-cookie values to protect their integrity. The encrypted cookie jar encrypts the
-values in addition to signing them, so that they cannot be read by the end-user.
-Refer to the [API documentation](https://api.rubyonrails.org/classes/ActionDispatch/Cookies.html)
-for more details.
+The `CookieStore`` can store 4 kB of data - much less than the other storage options - but this is usually enough. Storing large amounts of data in the session is discouraged. You should especially avoid storing complex objects (such as model instances) in the session.
+
+You can use the `CacheStore` if your sessions don't store critical data or don't need to be around for long periods (for instance if you just use the flash for messaging). This will store sessions using the cache implementation you have configured for your application. The advantage of this is that you can use your existing cache infrastructure for storing sessions without requiring any additional setup or administration. The downside, of course, is that the sessions will be ephemeral and could disappear at any time.
```

Comment:
I like @OughtPuts suggestion, but I've also remove "ofcourse" from the last sentence.

```suggestion
You can use the `CacheStore` if your sessions don't store critical data or don't need to be around for long periods (for instance if you just use the flash for messaging). This will store sessions using the cache implementation you have configured for your application. The advantage is that you can use your existing cache infrastructure for storing sessions without requiring any additional setup or administration. The downside is that the session storage will be temporary and they could disappear at any time.
```